docs/OV_Runtime_UG/Python_API_exclusives.md

```diff
@@ -141,3 +141,18 @@ The callback of `AsyncInferQueue` is uniform for every job. When executed, GIL i
     :fragment: [asyncinferqueue_set_callback]
 
 @endsphinxdirective
+
+@anchor releasing_gil
+### Releasing the GIL
+
+Some functions in Python API release the Global Lock Interpreter (GIL) while running work-intensive code. It can help you to achieve more parallelism in your application using Python threads. For more information about GIL please refer to the Python documentation.
+
+@sphinxdirective
+
+.. doxygensnippet:: docs/snippets/ov_python_exclusives.py
+    :language: python
+    :fragment: [releasing_gil]
+
+@endsphinxdirective
+
+> **NOTE**: While GIL is released functions can still modify and/or operate on Python objects in C++, thus there is no reference counting. User is responsible for thread safety if sharing of these objects with other thread occurs. It can affects your code only if multiple threads are spawned in Python.
```

Comment:
> While GIL is released ... operate on Python objects

- we can modify C++ state of OpenVINO objects - this doesn't break Python state protected by GIL.
- &#x2757; we can't query or modify state of any Python objects
- &#x2757; also we can't call "Python" callbacks (reacquire GIL before)

so this note is not accurate.

Reply:
For sure python objects in this meaning are not modified, but GIL is only protects reference counters in this case. But what if  C++ state of python object is modifying simultaneously in several python threads? It can be not thread safety and GIL doesn't protect it.